src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,654 @@
+  protection points within PPA and GWPZ protection areas. Post IAP grant EIRP
+  and interference contribution is calculated over each 5MHz channel
+
+  Args:
+    protection_point: List of protection points inside protection area
+    channels: List of tuples containing (low_freq, high_freq) of each 5MHz 
+              segment of the whole frequency range of protected entity
+    low_freq: low frequency of protection constraint (Hz)
+    high_freq: high frequency of protection constraint (Hz)
+    grant_objects: a list of grant requests, each one being a tuple 
+                   containing grant information
+    fss_info: contains information on antenna height and 
+              weights on the tangent and perpendicular components.
+    esc_antenna_info: contains information on ESC antenna height, azimuth, 
+                      gain and pattern gain
+    threshold: protection threshold (dBm/IAPBW_HZ)
+    protection_ent_type: an enum member of class ProtectedEntityType
+  Returns:
+    Updates global container aggregate_interference dictionary in the format of 
+    {(latitude,longitude,low_freq,high_freq,entity_type):aggregate_interference}
+    aggregate_interference value is in mW/RBW.
+    and global container post_iap_managed_grants nested dictionary in the format of
+    {grant_id:{ProtectionConstraint:PostIAPGrantInfo,ProtectionConstraint:PostIAPGrantInfo},
+    where ProtectionConstraint and PostIAPGrantInfo is a namedtuple 
+  """
+
+  for channel in channels:
+    # Set interference quota for protection_point, channel combination
+
+    # Calculate Roll of Attenuation for ESC Sensor Protection entity
+    if protection_ent_type is interf.ProtectedEntityType.ESC_CAT_A or\
+      protection_ent_type is interf.ProtectedEntityType.ESC_CAT_B:
+      if channel[0] >= 3650.e6:
+        center_freq = (channel[0] + channel[1]) / 2
+        interference_quota = interf.dbToLinear(interf.linearToDb(threshold) - 
+                (2.5 + ((center_freq - interf.ESC_CH21_CF_HZ) / interf.ONE_MHZ)))
+      else:
+        interference_quota = threshold
+
+    # Get the FSS blocking passband range
+    elif protection_ent_type is interf.ProtectedEntityType.FSS_BLOCKING:
+      interference_quota = threshold
+      channel = [low_freq, high_freq]
+
+    else:
+      interference_quota = threshold
+
+    # Get protection constraint over 5MHz channel range
+    channel_constraint = interf.ProtectionConstraint(latitude=protection_point[0],
+                           longitude=protection_point[1], low_frequency=channel[0],
+                           high_frequency=channel[1], entity_type=protection_ent_type)
+
+    # Identify CBSD grants in the neighborhood of the protection point 
+    # and channel
+    neighborhood_grants = interf.findOverlappingGrantsInsideNeighborhood(
+                            grant_objects, channel_constraint)
+    num_grants = len(neighborhood_grants)
+
+    if num_grants is 0:
```

Comment:
Bad construct. This works in CPython because of internal python optimization which keeps singleton for small numbers, but cannot be insured for other python implementation,.

Use instead:
```
if not neighborhood_grants:
```

Reply:
Updated